packages/i18next/src/register-discordjs.ts

```diff
@@ -0,0 +1,134 @@
+		 */
+		fetchLanguage(): Promise<string>;
+
+		/**
+		 * Function that gets a TFunction (translator function) from i18next.
+		 * @since 1.0.0
+		 * @return An i18next TFunction.
+		 */
+		fetchT(): Promise<TFunction>;
+
+		/**
+		 * Function that resolves a language key from the store.
+		 * @since 1.0.0
+		 * @return A string, which is the translated result of the key, with templated values.
+		 */
+		fetchLanguageKey(key: string, ...values: readonly any[]): Promise<string>;
+
+		/**
+		 * Function that sends a message to the context channel with the translated key and values.
+		 * Functionally equivalent to piping fetchLanguageKey through channel#send.
+		 * @since 1.0.0
+		 * @return The message object that was sent.
+		 */
+		sendTranslated(
+			key: string,
+			values?: readonly unknown[],
+			options?: MessageOptions | (MessageOptions & { split?: false }) | MessageAdditions
+		): Promise<Message>;
+		sendTranslated(key: string, values?: readonly unknown[], options?: MessageOptions & { split: true | SplitOptions }): Promise<Message[]>;
+		sendTranslated(key: string, options?: MessageOptions | (MessageOptions & { split?: false }) | MessageAdditions): Promise<Message>;
+		sendTranslated(key: string, options?: MessageOptions & { split: true | SplitOptions }): Promise<Message[]>;
+		sendTranslated(
+			key: string,
+			valuesOrOptions?: readonly unknown[] | MessageOptions | MessageAdditions,
+			rawOptions?: MessageOptions
+		): Promise<Message | Message[]>;
+	}
+
+	export interface Client {
+		i18n: I18nextHandler;
+
+		/**
+		 * The method to be overriden by the developer.
+		 * Note: In the event that fetchLanguage is not defined or returns null or undefined
+		 * the order of defaulting will be as follows:
+		 * client.fetchLanguage -> message.guild.preferredLocale -> this.client.options.i18n.defaultName -> 'en-US'
+		 * @since 1.0.0
+		 * @return A string for the desired language or null for no match.
+		 * @example
+		 * ```typescript
+		 * // Always use the same language (no per-guild configuration):
+		 * client.fetchLanguage = () => 'en-US';
+		 * ```
+		 * @example
+		 * ```typescript
+		 * // Retrieving the prefix from an SQL database:
+		 * client.fetchLanguage = async (message) => {
+		 *   const guild = await driver.getOne('SELECT language FROM public.guild WHERE id = $1', [message.guild.id]);
+		 *   return guild?.language ?? 'en-US';
+		 * };
+		 * ```
+		 * @example
+		 * ```typescript
+		 * // Retrieving the language from an ORM:
+		 * client.fetchLanguage = async (message) => {
+		 *   const guild = await driver.getRepository(GuildEntity).findOne({ id: message.guild.id });
+		 *   return guild?.language ?? 'en-US';
+		 * };
+		 * ```
+		 */
+		fetchLanguage: (message: any) => Promise<string | null> | string | null;
+	}
+
+	export interface ClientOptions {
+		i18n?: I18nOptions;
```

Comment:
Some documentation here would be nice (also requested by Kyra when Q was doing the PR for this line)